stdlib/Profile/src/Profile.jl

```diff
@@ -403,9 +422,9 @@ end
 
 function has_meta(data)
     for i in 6:length(data)
-        data[i] == 0 || continue            # first block end null
-        data[i - 1] == 0 || continue        # second block end null
-        data[i - META_OFFSET_SLEEPSTATE] in 1:2 || continue
+        data[i] == 0 || continue                            # first block end null
+        data[i - 1] == 0 || continue                        # second block end null
+        data[i - META_OFFSET_SLEEPSTATE] in 1:3 || continue # 1 for not sleeping, 2 for sleeping, 3 for task profiler fake state
```

Comment:
@d-netto do we need to update PProf.jl and the other profile consumer tools to be aware of state 3?

Reply:
I think it would be helpful if this comment either indicated where to go read the original definition of those states, or provided more info about them.

Reply:
Possibly, though I find only a single reference to `META_OFFSET_SLEEPSTATE` in `PProf.jl`.

Changing `thread_sleeping != 0` to `thread_sleeping == 1` [here](https://github.com/JuliaPerf/PProf.jl/blob/663b96ede57e507bc059e99db564f8e0fce4de98/src/PProf.jl#L185) seems enough for `Proof.jl` at least?

Reply:
@NHDaly, what other profiler consumer tools do we have?

Reply:
https://github.com/timholy/FlameGraphs.jl

Reply:
And the VSCode profile visualizer.

Actually, there's a whole, long list here:
- https://docs.julialang.org/en/v1/manual/profile/#Basic-usage

But i think if you check PProf, VSCode, and ProfileView (which uses FlameGraphs) that'll be enough

Reply:
Looks like nothing else uses it, https://github.com/search?q=META_OFFSET_SLEEPSTATE+NOT+is%3Afork&type=code

Reply:
X-ref: https://github.com/JuliaPerf/PProf.jl/pull/101.